### DISCUSSION TITLE
Cant figure out how to use this to connect my STM32F072RB to the NHD-4.3-480272FT-CTXL-CTP ,while using the ArduinoIDE to code the STM

### DISCUSSION THREAD

**Discussion by prats365** [RudolphRiedel/FT800-FT813#92] [Q&A] [answered]
I'm very new to this , and I'm struggling to understand the documentation well enough to figure out how to do it. Same reason why i'm using the arduino ide to try and code the stm as cube ide was a little too hard to directly start out with.

**Comment by RudolphRiedel** [comment]
Well, I am not really supporting the Arduino IDE so that part is not actually documented on my side and things are bit messy.
I highly recommend using VSCode and PlatformIO, especially for Arduino programming.
One of the many reasons is that PlatformIO allows to set defines in the build environment and therefore allows to configure a library without the need to change the code itself.

I do happen to have an Arduino IDE compatible demo project though and I just modified it a little:
[EVE_Test_Arduino_IDE.zip](https://github.com/RudolphRiedel/FT800-FT813/files/11898334/EVE_Test_Arduino_IDE.zip)

This has the latest files from the repository.
EVE_config.h is modified with a macro define in line 41 for the display to be used.

The target configuration used is generic Arduino as I do not have a target for the STM32F072RB specifically.
The pins to be used are the SPI default pins for SCK, MOSI and MISO.
And in EVE_target/EVE_target_Arduino_generic.h the chip-select and power-down pins are defined:

``
#if !defined(EVE_CS)
#define EVE_CS 10
#endif

#if !defined(EVE_PDN)
#define EVE_PDN 8
#endif
``

Note, this file is not visible directly when openeing the .ino in the Arduino IDE and you can not use Sketch/Add File as this would copy the file to the parent folder but it would not be used when compiling the project.

I just ran Arduino IDE 1.8.19, installed https://github.com/stm32duino/Arduino_Core_STM32.
I selected board "Nucleo-64" and "Nucleo-F027RB" and hit "verify".
And the project just compiled without a single warning.

I am quite positive that it will work, although a bit slower as the controller would allow due to a lack of support for DMA in the vanilla Arduino SPI class and code in my library to work around this for the STM32F072RB.

Hmm, and it builds for the Nucleo-C031C6 as well, nice.
Ah, it was added in v2.5.0 and PlatformIO is still at v2.4.0.

**Reply by prats365** [reply]
Hi , yeah turns out i cant use the arduino ide for the project i'm trying to do so I shifted back to CubeIDE , but since VSC is also an option i'll try  that out too. Thanks again.

And i think i'll try the arduino implementation one last time haha.

Also it wont work with ARduinoIDE 2.1.1 i assume , so i'll jjust downgrade to 1.8

**Reply by prats365** [reply]
Also I'm not very sure as to how and where I should modify the code for my board so that I can change the required pins for my SPI use.

**Comment by RudolphRiedel** [comment]
I am not really using CubeIDE, at least not beyond configuring the pins and generating init code.
Well, I am not really using STM32 controllers so far.

In CubeIDE this can be done thru the project properties:
![grafik](https://github.com/RudolphRiedel/FT800-FT813/assets/31180093/c3e19bcc-15e1-43f4-b475-450366d9e59a)

However, as I wrote above, there is no specific STM32F072RB in my code so far.
And the non-Arduino side sure is messy so far.

Here is a test project I did back in march for a board I designed:
[EVE_Test_STM32G0B1_RVT50H.zip](https://github.com/RudolphRiedel/FT800-FT813/files/11959802/EVE_Test_STM32G0B1_RVT50H.zip)

This was merely a test for the board and not an actual project.
But as a project this is more complete and was working.

In contrast to this one:
https://github.com/RudolphRiedel/FT800-FT813/tree/5.x/examples/EVE_Test_STM32_RiTFT50_PlatformIO

This does compile for a bunch of STM32 but does not actually do anything as the main.c has no code to actually configure the controller.
So this is more proof of concept for now that my library code in general compiles for STM32, down to using the correct LL_HAL SPI functions.

The configuration is done in the platformio.ini file.

There already is this:
[env:STM32F030]
board = nucleo_f030r8
build_flags =
            ${env.build_flags}
            -D STM32F0

So the STM32F0 family already is supported.

Changing it like this:
[env:STM32F072]
board = nucleo_f072rb
build_flags =
            ${env.build_flags}
            -D STM32F0

And the simple example compiles fine, HAL driver and everything else.

[env:STM32F072]
board = nucleo_f072rb
build_flags =
            ${env.build_flags}
            -D STM32F0

Ugh, adding the pins to platformio.ini does not work though.
So for now this requires to change  EVE_target/EVE_target_STM32.h - which is bad for working with PlatformIO as the library is downloaded by platformio and not something that should need modifying.
Ok, added to the to-do list...

**Reply by prats365** [reply]
So what you're saying is , even if I do use VSC with PlatformIO (while using the Arduino framework, there's still a chance it might not work properly ? Sorry if the question's are a tad bit dumbed down , not really sure about my  course of action :(

**Reply by RudolphRiedel** [reply, marked as answer]
Ah, still using the Arduino framework.
That should be far easier since the part that is missing from non-Arduino example is the controller initialization including SPI and pins, but the Arduino framework already takes care of that.

Use this as a starting point:
https://github.com/RudolphRiedel/FT800-FT813/blob/5.x/examples/EVE_Test_Arduino_PlatformIO/platformio.ini

Then in platformio.ino:
[env:STM32F030]
platform = ststm32
board = nucleo_f030r8
upload_protocol = stlink
build_unflags = -Os
build_flags =
        ${env.build_flags}
         -D EVE_CS=10
         -D EVE_PDN=8
         -O2

I only can't confirm this right now, I need to leave for work.

**Reply by prats365** [reply]
Thanks for the quick responses man , you're a life saver. I'll give that suggestion a try.
